File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var stack = []
    var i = 0;
    while (i < s.length) {
        if (stack.length > 0 && stack[stack.length - 1] == s[i]) {
            stack.pop();
        } else {
            stack.push(s[i]);
        }
        i += 1;
    }
    return stack.length == 0 ? 1 : 0;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []  # 스택 초기화
    for i in range(len(s)):
        if stack and stack[-1] == s[i]:
            stack.pop()
        else:
            stack.append(s[i])
            
    if stack:
        return 0
    
    return 1